packages/ui-components/style/rjsfTemplates.css

```diff
@@ -115,19 +115,34 @@
   border: 1px solid var(--jp-border-color2);
   border-radius: 4px;
   margin: 4px;
+  position: relative;
 }
 
 .jp-ArrayOperations {
   display: flex;
   margin-left: 8px;
 }
 
-.jp-ArrayOperationsButton {
+.field-array-of-object .jp-ArrayOperations {
+  position: absolute;
+  right: 2px;
+  top: 10px;
+}
+
+/* Display the array operations toolbar on hover only for array of objects */
+.jp-arrayFieldWrapper
+  .field-array-of-object
+  .array-item:not(:hover)
+  .jp-ArrayOperations {
+  display: none;
+}
```

Comment:
I think there would be some accessibility considerations - probably it should also show up when the field is focus-in